techmatters/terraso-mobile-client
# feat: Add automatic pull mechanism

## Description
- `PullRequester` will request (but not execute) a pull. It will update redux state in a new `sync` slice to note that a pull is now requested.
- `PullDispatcher` will decide to do a pull when a pull has been requested, and if it detects all the other conditions are right.

### Related Issues
Implements #2291 and #2292 (which, apologies, should really be one issue)

### Verification steps
- Pull should happen on open, when app goes from offline to online (with 500ms debounce), following a push that returned errors, and every 5 minutes.
- Pull should not happen before login, when offline, when there are local changes that have not yet been pushed, or when the app is not active in the foreground.
